packages/animations/browser/src/dsl/animation_transition_factory.ts

```diff
@@ -91,13 +92,65 @@ export class AnimationTransitionFactory {
       }
     });
 
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
+      checkNonAnimatableInTimelines(timelines, this._triggerName, driver);
+    }
+
     const queriedElementsList = iteratorToArray(queriedElements.values());
     return createTransitionInstruction(
         element, this._triggerName, currentState, nextState, isRemoval, currentStateStyles,
         nextStateStyles, timelines, queriedElementsList, preStyleMap, postStyleMap, totalTime);
   }
 }
 
+/**
+ * Checks inside a set of timelines if they try to animate a css property which is not considered
+ * animatable, in that case it prints a warning on the console.
+ * Besides that the function doesn't have any other effect.
+ *
+ * Note: this check is done here after the timelines are built instead of doing on a lower level so
+ * that we can make sure that the warning appears only once per instruction (we can aggregate here
+ * all the issues instead of finding them separately).
+ *
+ * @param timelines The built timelines for the current instruction.
+ * @param triggerName The name of the trigger for the current instruction.
+ * @param driver Animation driver used to perform the check.
+ *
+ */
+function checkNonAnimatableInTimelines(
+    timelines: AnimationTimelineInstruction[], triggerName: string, driver: AnimationDriver): void {
+  if (!driver.validateAnimatableStyleProperty) {
+    return;
+  }
+
+  const invalidNonAnimatableProps = new Set<string>();
+
+  timelines.forEach(({keyframes}) => {
+    const nonAnimatablePropsFirstValues = new Map<string, string|number>();
+    keyframes.forEach(keyframe => {
+      for (const [prop, value] of keyframe.entries()) {
+        if (!driver.validateAnimatableStyleProperty!(prop)) {
```

Comment:
I'm wondering if the non-null assertion is needed here (since we exit the function if it doesn't exist)?
```suggestion
        if (!driver.validateAnimatableStyleProperty(prop)) {
```

Reply:
Yes, it is pretty strange... but typescript for some reason doesn't seem able to infer that :confused: 

![Screenshot at 2022-07-12 20-54-30](https://user-images.githubusercontent.com/61631103/178583696-e2931bc0-f707-48e2-9423-932c5528d60d.png)

Not even reversing the order of the if works:
![Screenshot at 2022-07-12 20-56-24](https://user-images.githubusercontent.com/61631103/178583778-f00e57cc-df4b-4783-8fc8-87c7f0c8998a.png)

One thing that does seem to do the trick is using a temporary variable for the function:
![Screenshot at 2022-07-12 21-01-01](https://user-images.githubusercontent.com/61631103/178583892-52b7080e-a5c2-4d23-9d2a-cb109c7ab34e.png)

what do you think?

Reply:
Thanks for looking into this. No worries, let's just keep the non-null assertion :)

Reply:
Ah ok thanks :slightly_smiling_face: 

But it is indeed pretty annoying.... I wonder why typescript doesn't infer the thing :cry:

Reply:
@AndrewKushnir seems like the inference breaks inside the arrow functions:
![Screenshot at 2022-07-12 22-07-59](https://user-images.githubusercontent.com/61631103/178596432-0c608f0c-bc42-4d42-8d35-912a21bdaca6.png)

So another alternative could be to use `for-of`s instead of `forEach`s:
![Screenshot at 2022-07-12 22-12-22](https://user-images.githubusercontent.com/61631103/178596529-1f4b2c4c-933b-4eed-aeb3-acea10e16b39.png)

What do you think? does this look better?